compiled_types.ts

```diff
@@ -54,6 +66,32 @@ export type AuthenticationMetadata = DistributiveOmit<
   postSetup?: PostSetupMetadata[];
 };
 
+/** @hidden */
+type AuthenticationToMetadata<T extends Authentication> = DistributiveOmit<
+  T,
+  'getConnectionName' | 'getConnectionUserId' | 'postSetup'
+> & {
+  getConnectionName?: MetadataFormulaMetadata;
+  getConnectionUserId?: MetadataFormulaMetadata;
+  postSetup?: PostSetupMetadata[];
+};
+
+/** @hidden */
+export interface AuthenticationMetadataTypeMap {
```

Comment:
While you're here, mind also just adding a sibling type map that maps from the type to the full interface, e.g. `[AuthenticationType.QueryParamToken]: QueryParamTokenAuthentication;` just so we have that for future convenience?

Reply:
Added in types.ts, which seemed like the logical place